spyder/plugins/ipythonconsole/widgets/client.py

```diff
@@ -101,28 +101,21 @@ def __init__(self, parent, id_,
                  options_button=None,
                  show_elapsed_time=False,
                  reset_warning=True,
-                 ask_before_restart=True,
-                 ask_before_closing=False,
                  css_path=None,
-                 handlers={},
-                 stderr_obj=None,
-                 stdout_obj=None,
-                 fault_obj=None):
```

Comment:
connection_file and *_obj are bundled together with everything needed to interface with the kernel